src/views/Room.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
+import './Room.css';
+import Constants from '../constants/Constants';
+import * as Types from '../constants/Types';
+import socket from '../constants/socket-context';
+import Messaging from './Messaging';
+import FileTransfer from './FileTransfer';
+import PreConnectionRoom from '../components/PreConnectionRoom';
+import ConnectedRoom from '../components/ConnectedRoom';
+
+type RoomProps = {
+    currentRoom: Types.Room;
+    displayName: Types.UserDisplay;
+    onInitialMessageSend: Function;
+    onInitialFileSend: Function;
+    addRoomMessage: Function;
+    updateCompletedFile: Function;
+};
+
+type RoomState = {};
+
+class Room extends Component<RoomProps, RoomState> {
+    componentDidMount(): void {
+        // Attempt to connect to room when this component loads.
+        socket.emit(Constants.CONNECT_TO_ROOM, { roomid: this.props.currentRoom.roomid });
+    }
+
+    render(): React.ReactNode {
+        return (
+            <div className="room-container standard-container-padding">
+                {!this.props.currentRoom.requestSent ? (
+                    <PreConnectionRoom currentRoom={this.props.currentRoom} displayName={this.props.displayName} />
+                ) : (
+                    <ConnectedRoom currentRoom={this.props.currentRoom} displayName={this.props.displayName} />
+                )}
+                {/* <FileTransfer
```

Comment:
Removed filetransfer for now, and only focused on messaging bc it was buggy when both were on the mainhome page, but I think when all the rtc logic is moved into redux, and both messaging and file transfer reference the same rtconnection, i think this will be resolved.

Reply:
Sounds good, will attempt this.